### PR TITLE
Check candidate signatures before including them in `set_heads`

### PR DESCRIPTION
**Pull request by bkchr**
This work around the bug described in: https://github.com/paritytech/polkadot/issues/1327

We check the candidate signatures in `create_inherent` to ensure that
all are valid. If one isn't valid, we don't include any candidate for
the current inherent.

### CODE REVIEW

File: runtime/common/src/parachains.rs [hunk 1]
```diff
@@ -1509,7 +1541,13 @@ impl<T: Trait> ProvideInherent for Module<T> {
 			.expect("Parachain heads could not be decoded.")
 			.expect("No parachain heads found in inherent data.");
 
-		Some(Call::set_heads(data))
+		// Temporary solution for:
+		// https://github.com/paritytech/polkadot/issues/1327
+		if Self::check_candidates_signatures(&data).is_ok() {
```
Comment (rphmeier) [inline runtime/common/src/parachains.rs:1546]: Could this not just do `Module::<T>::set_heads(Origin::none(), ...).is_ok()` without the other changes?

Comment (bkchr) [reply runtime/common/src/parachains.rs:1546]: Yes this would work as well. But set heads is doing much more and we don't really need this here IMHO. This just wastes resources.

**Comment by gavofyork** [comment]
will need a bump to each runtime

**Comment by bkchr** [comment]
> will need a bump to each runtime

Done

**Comment by gavofyork** [comment]
> We check the candidate signatures in create_inherent to ensure that
all are valid. If one isn't valid, we don't include any candidate for
the current inherent.

Why is the inherent even valid in this case? Shouldn't an invalid inherent result in an invalid block?

**Comment by bkchr** [comment]
> > We check the candidate signatures in create_inherent to ensure that
> > all are valid. If one isn't valid, we don't include any candidate for
> > the current inherent.
> 
> Why is the inherent even valid in this case? Shouldn't an invalid inherent result in an invalid block?

Yes and that is done, but this means we always run into the same problem that is described here: https://github.com/paritytech/polkadot/issues/1327

TLDR the new block changes the authority set, we check based on the authority set the signatures of the PoVBlock's. As the authorities (may) changed, the signature check will fail and this brings us in a loop where this always happens and the network is down.
